src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Extensions/DiagnosticAnalyzerExtensions.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+namespace Microsoft.CodeAnalysis.Diagnostics
+{
+    internal static class CompilerDiagnosticAnalyzerNames
+    {
+        internal const string CSharpCompilerAnalyzerTypeName = "Microsoft.CodeAnalysis.Diagnostics.CSharp.CSharpCompilerDiagnosticAnalyzer";
+        internal const string VisualBasicCompilerAnalyzerTypeName = "Microsoft.CodeAnalysis.Diagnostics.VisualBasic.VisualBasicCompilerDiagnosticAnalyzer";
+    }
+
+    internal static partial class DiagnosticAnalyzerExtensions
+    {
+        public static bool IsCompilerAnalyzer(this DiagnosticAnalyzer analyzer)
```

Comment:
❕ In the first pass, we should look at `SupportedDiagnostics` to see if any diagnostic descriptors have the tag `WellKnownDiagnosticTags.Compiler`.

Reply:
This is existing code that was just moved. Note that we explicitly avoid invoking SupportedDiagnostics on compiler analyzer for performance reasons. We do not want to realize thousands of descriptors here.